nix/package-settings.nix

```diff
@@ -0,0 +1,102 @@
+# A haskell-flake module providing a simpler interface to `overrides`. The
+# config implementation simply expands to `overrides`.
+{ config, pkgs, lib, ... }:
+
+{
+  options = {
+    packageSettings = lib.mkOption {
+      type = lib.types.lazyAttrsOf (lib.types.submodule {
+        options = {
+          input = lib.mkOption {
+            default = { };
+            type = lib.types.deferredModuleWith {
+              staticModules = [
+                ({ lib, self, super, ... }: {
+                  # TODO: What is the best way to do enum type here?
```

Comment:
Normally there is `types.oneOf`, but I can't use it here because these are top-level static module options. Is there a better way of doing this?

Reply:
If we can do the top-level `{self, super, ...}` as indicated in the above comment, I could simply make `input` an `oneOf` in the style of https://github.com/NixOS/nixpkgs/blame/995edc972ad3a1e291ac22d74b9610821357175f/pkgs/development/haskell-modules/lib/compose.nix#L44-L54